roles/feature-acm/tasks/post-deploy.yaml

```diff
@@ -0,0 +1,129 @@
+# This deploys RH ACM Operator to a cluster
+# Based on:
+# https://docs.redhat.com/en/documentation/red_hat_advanced_cluster_management_for_kubernetes/2.14/html/install/installing#installing-from-the-cli
+
+- name: Ensure cluster is online
+  wait_for:
+    host: "api.{{ inventory_hostname }}.{{ cluster_base_domain }}"
+    port: 6443
+    sleep: 1
+    timeout: 10
+
+- name: Fetch kubeconfig from vault
+  ansible.builtin.include_role:
+    name: internal-fetch-kubeconfig
+
+# --------------------------------------------
+# ----- ACM Operator
+# --------------------------------------------
+- name: Install ACM operator
+  register: acm_operator
+  kubernetes.core.k8s:
+    kubeconfig: "{{ cluster_access_kubeconfig }}"
+    state: present
+    definition:
+      - apiVersion: v1
+        kind: Namespace
+        metadata:
+          name: open-cluster-management
+      - apiVersion: operators.coreos.com/v1
+        kind: OperatorGroup
+        metadata:
+          name: acm-operators
+          namespace: open-cluster-management
+        spec:
+          targetNamespaces:
+          - open-cluster-management
+      - apiVersion: operators.coreos.com/v1alpha1
+        kind: Subscription
+        metadata:
+          name: acm-operator-subscription
+          namespace: open-cluster-management
+        spec:
+          channel: "{{ acm_operator_channel }}"
+          name: advanced-cluster-management
+          source: redhat-operators
+          sourceNamespace: openshift-marketplace
+          installPlanApproval: Automatic
+
+- name: Wait for operator pod to be ready
+  kubernetes.core.k8s_info:
+    kubeconfig: "{{ cluster_access_kubeconfig }}"
+    kind: Pod
+    namespace: open-cluster-management
+    label_selectors:
+      - "name = multiclusterhub-operator"
+    wait: yes
+    wait_timeout: 300
+    wait_condition:
+      type: Ready
+      status: True
+
+- name: Install MultiClusterHub (Coffee Time!)
+  kubernetes.core.k8s:
+    kubeconfig: "{{ cluster_access_kubeconfig }}"
+    state: present
+    wait: yes
+    wait_timeout: 1200
+    wait_condition:
+      type: Complete
+      status: True
+    definition:
+    - apiVersion: operator.open-cluster-management.io/v1
+      kind: MultiClusterHub
+      metadata:
+        name: multiclusterhub
+        namespace: open-cluster-management
+      spec:
+        availabilityConfig: Basic
+        enableClusterBackup: false
+        ingress: {}
+        localClusterName: local-cluster
+        overrides:
+          components:
```

Comment:
Have you tried just to add the components with your values? Because this list might chnage from ACM release to ACM release.

Reply:
good point. let me move this out of the task into a proper template, then we can better surrond / protect with if blocks....